### ISSUE TITLE
Create documentation explaining CSharpier formatting decisions

### ISSUE THREAD

**Issue by belav** [belav/csharpier#785]
### Discussed in https://github.com/belav/csharpier/discussions/778

<div type='discussions-op-text'>

<sup>Originally posted by **andrewrondeau-optirtc** December 27, 2022</sup>
Hi, apologies if this is a noob question!

Do you have any documentation that **explicitly** explains the style that CSharpier applies? (I like that it's opinionated, BTW.)

Even though I've run CSharpier on my codebase, I'd like to be able to read through a general description of the kinds of changes it makes to my code.

If I choose to integrate this into the workflow, it's a lot easier to explain it if I can point out specific style changes it will make, and changes it won't make.</div>

**Comment by commonquail** [comment]
This is an area where google-java-format excels because of https://github.com/google/google-java-format/wiki/The-Rectangle-Rule and Prettier in comparison is an inconsistent mess. Faulty implementation notwithstanding, numerous presently open inconsistencies in and questions about CSharpier's behaviour would be neatly resolved in theory by adhering to the same principle as google-java-format.

I understand that CSharpier draws inspiration from Prettier. I am also saying in the same breath that Prettier needlessly faces several of these difficulties and does not represent the state of the art. For another perspective, rustfmt is somewhere in-between, condensing some things google-java-format would expand and expanding some things google-java-format would condense.

In chronological order, at least the following presently open issues raise questions about intended or desired behaviour; notably, many propose a solution that aligns with "the rectangle rule":

- #108
- #205
- #238
- #243
- #290
- #292
- #348
- #356
- #449
- #450
- #451
- #452
- #468
- #469
- #672
- #765
- #766
- #792
- #803
- #836

**Comment by commonquail** [comment]
> I think you may have linked to this in the past?

Yeah, that was me. I didn't really expect a reply. It's not like google-java-format doesn't have its own issues, including some caused by that principle, but it's the only tool I know of that has a fairly consistent style that is also explained by principle -- I've found that useful for comprehension but of course it helps that I also like the resulting style.

I'll add for the record, too, that none of this is meant as criticism and I still choose CSharpier. I just indexed the items that seemed to represent incidental (_possibly_ unwanted) behaviour.

> I don't suppose you know of a playground for google-java-format?

No, I've never come across a such. It's a self-contained jar file, however, so it's straightforward to run by hand, and there is https://plugins.jetbrains.com/plugin/8527-google-java-format.
